src/XPurchase/VirtoCommerce.XPurchase/CartAggregate.cs

```diff
@@ -199,7 +199,64 @@ await AddItemAsync(new NewCartItem(item.ProductId, item.Quantity)
             }
 
             return this;
-        }
+        }
+
+        public virtual async Task<CartAggregate> AddGiftItemsAsync(IReadOnlyCollection<string> productIds, IReadOnlyCollection<LineItem> availableGifts)

```

Comment:
I think this method must accept [GitftReward](https://github.com/VirtoCommerce/vc-module-marketing/blob/dev/src/VirtoCommerce.MarketingModule.Core/Model/Promotions/Rewards/GiftReward.cs)  as an argument to be able to support gifts without products